karate-core/src/main/java/com/intuit/karate/core/ScenarioResult.java

```diff
@@ -106,7 +107,19 @@ private static void recurse(List<Map> list, StepResult stepResult, int depth) {
                 call.setLine(stepResult.getStep().getLine());
                 call.setPrefix(StringUtils.repeat('>', depth));
                 call.setText(fr.getCallNameForReport());
-                call.setDocString(fr.getCallArgPretty());
+                boolean isShowCallArg= true;
+                for(ScenarioResult scenarioResult: fr.getScenarioResults()){
+                    for (StepResult sr: scenarioResult.getStepResults()){
+                        if(sr.isHidden()){
+                            Map<String, Object> callArgMap = fr.getCallArg();
+                            isShowCallArg=false;
```

Comment:
I need time to dig into this now, because this doesn't seem right. the decision to hide a call argument should be at the scenario level and not based on any one or two steps. if it means we capture the fact if a scenario is hidden (e.g. via a tag or config) on the scenario result, so be it.

I'm sorry I can't spend time giving you more leads or instructions, you are free to abandon all changes if you feel so

Reply:
I certainly think this PR has to be re-done because 8 commits for a few lines change does not make sense, or maybe I need to figure out how to "squash" changes which I normally don't do or maybe my lack of understanding